pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -712,6 +712,8 @@ type Git struct {
 	Operator Operator `json:"operator,omitempty"`
 	// +optional
 	BootstrapProfile *Profile `json:"bootstrapProfile,omitempty"` // one or many profiles to enable on this cluster once it is created
+	// +optional
+	CommitOperatorManifests *bool `json:"commitOperatorManifests,omitempty"` // Commit and push Flux manifests to the Git Repo on install
```

Comment:
Is there any particular reason to add this new field?

Reply:
Yeah. I believe it has some relationship with the way how I setup Flux.

I usually point multiple clusters' Flux to the same config repository, so that I can swap the cluster for e.g. blue-green-style K8s version upgrade easily(I don't need to lock, recreate or copy the config repo for the new cluster. I can just reuse the config repo this way).

It's like you reuse AWS launch configuration when you want two identical ASGs. I want two identical clusters so I reuse the config repo.

Trying to generate, commit, and pushing Flux manifests for each cluster creation breaks this model.

I thought some percentage of people who would benefit from the unattended Flux installation would use the same model as mine, and therefore find this helpful. That's why I included it in this PR.

WDYT?

Reply:
After consulting with the flux team it seems like a valid use case for other people too so let's keep this flag. I think it makes more sense inside the Operator as `git.operator.commitManifests`, WDYT?

Reply:
@martina-if Thanks! I believe this should better be here as-is, or alternatively under a new field `git.installer` like `git.installer.commitManifests`.

I'm thinking about it this way because `git.operator` you've suggested seems to be dedicated for configuring the operator(flux), not the installer of the operator. And that's the installer who commit/push Flux manifests.

From what I can see from the current spec, anything directly under `git` seems to be for configuring the installer, which make me thing this can safely be `git.commitManifests`, not something new like e.g. `git.installer.commitManiefsts`. But I'm definitely open to more feedback/comments regarding this. WDYT?

Reply:
Or perhaps it can be `git.operator.commitManifests` because it's for commiting manifests for the operator? If that's your intention, that makes sense. Would you mind confirming?

Reply:
Exactly, that's what I meant :)

Reply:
Thanks! Addressed in d14cb1a6